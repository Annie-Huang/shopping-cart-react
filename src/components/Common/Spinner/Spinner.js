import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import { Dimmer, Loader } from 'semantic-ui-react'


export class Spinner extends Component{
    render() {
        return (
            this.props.isLoading ?
                <Dimmer active inverted>
                    <Loader size='large' active>Loading</Loader>
                </Dimmer>
                : null
        )
    }
}

Spinner.propTypes = {
    isLoading: PropTypes.bool.isRequired
};

const mapStateToProps = (state) => ({
    isLoading: state.ajaxCallsInProgress > 0
});

export default connect(mapStateToProps, null)(Spinner);
