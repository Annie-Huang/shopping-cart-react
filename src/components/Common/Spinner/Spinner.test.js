import React from "react";
import {shallow} from "enzyme/build";
import {Spinner} from "./Spinner";


describe('Spinner', () => {
    it('should render Spinner component when loading', () => {
        const props = {
            isLoading: true
        };
        expect(shallow(<Spinner {...props} />)).toMatchSnapshot();
    });

    it('should render Spinner component when not loading', () => {
        const props = {
            isLoading: false
        };
        expect(shallow(<Spinner {...props} />)).toMatchSnapshot();
    });
});
