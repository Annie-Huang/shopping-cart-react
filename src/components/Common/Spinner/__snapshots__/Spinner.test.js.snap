// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render Spinner component when loading 1`] = `
<Dimmer
  active={true}
  inverted={true}
>
  <Loader
    active={true}
    size="large"
  >
    Loading
  </Loader>
</Dimmer>
`;

exports[`Spinner should render Spinner component when not loading 1`] = `""`;
